utilities/rpp-unittests/HOST/Single_host.cpp

```diff
@@ -0,0 +1,515 @@
+#include <time.h>
+using namespace cv;
+using namespace std;
+#include <CL/cl.hpp>
+#define images 100
+int G_IP_CHANNEL = 3;
+int G_MODE = 1;
+char src[1000] = {"/home/neel/Ulagammai/Input_Images_16/RGB"};
+char src_second[1000] = {"/home/neel/Ulagammai/Input_Images_16/RGB1"};
+char dst[1000] = {"/home/neel/Ulagammai/Output"};
+char funcType[1000] = {"Single"};
+
+int main(int argc, char **argv)
+{
+    int ip_channel = G_IP_CHANNEL;
+    int mode = G_MODE;
+    char *funcName = argv[1];
+    if (mode == 0)
+    {
+        strcat(funcType, "_CPU");
+    }
+    else if (mode == 1)
+    {
+        strcat(funcType, "_GPU");
+    }
+    else
+    {
+        strcat(funcType, "_HIP");
+    }
+    if (ip_channel == 1)
+    {
+        strcat(funcType, "_PLN");
+    }
+    else
+    {
+        strcat(funcType, "_PKD");
+    }
+
+    int i = 0, j = 0;
+    int minHeight = 30000, minWidth = 30000, maxHeight = 0, maxWidth = 0;
+    unsigned long long count = 0;
+    unsigned long long ioBufferSize = 0;
+
+    static int noOfImages = 0;
+
+    Mat image, image_second;
+
+    struct dirent *de;
+    char src1[1000];
+    strcpy(src1, src);
+    strcat(src1, "/");
+    char src1_second[1000];
+    strcpy(src1_second, src_second);
+    strcat(src1_second, "/");
+    strcat(funcName, funcType);
+    strcat(dst, "/");
+    strcat(dst, funcName);
+    mkdir(dst, 0700);
+    strcat(dst, "/");
+
+    DIR *dr = opendir(src);
+    while ((de = readdir(dr)) != NULL)
+    {
+        if (strcmp(de->d_name, ".") == 0 || strcmp(de->d_name, "..") == 0)
+            continue;
+        noOfImages += 1;
+        break;
+    }
+    closedir(dr);
+
+    RppiSize *srcSize = (RppiSize *)calloc(noOfImages, sizeof(RppiSize));
+    RppiSize *dstSize = (RppiSize *)calloc(noOfImages, sizeof(RppiSize));
+    //const int images = noOfImages;
+    char imageNames[images][1000];
+
+    DIR *dr1 = opendir(src);
+    while ((de = readdir(dr1)) != NULL)
+    {
+        if (strcmp(de->d_name, ".") == 0 || strcmp(de->d_name, "..") == 0)
+            continue;
+        strcpy(imageNames[count], de->d_name);
+        char temp[1000];
+        strcpy(temp, src1);
+        strcat(temp, imageNames[count]);
+        if (ip_channel == 3)
+        {
+            image = imread(temp, 1);
+        }
+        else
+        {
+            image = imread(temp, 0);
+        }
+        srcSize[count].height = image.rows;
+        srcSize[count].width = image.cols;
+        ioBufferSize += (unsigned long long)srcSize[count].height * (unsigned long long)srcSize[count].width * (unsigned long long)ip_channel;
+
+        count++;
+        break;
+    }
+    closedir(dr1);
+
+    Rpp8u *input = (Rpp8u *)calloc(ioBufferSize, sizeof(Rpp8u));
+    Rpp8u *input_second = (Rpp8u *)calloc(ioBufferSize, sizeof(Rpp8u));
+    Rpp8u *output = (Rpp8u *)calloc(ioBufferSize, sizeof(Rpp8u));
+
+    /* Read the input image */
+    DIR *dr2 = opendir(src);
+    DIR *dr2_second = opendir(src_second);
+    count = 0;
```

Comment:
Codacy found an issue: [Variable 'count' is reassigned a value before the old one has been used.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/rpp/pullRequest?prid=5976402)